src/lib/Type.hs

```diff
@@ -50,11 +49,21 @@ getType x = ignoreExcept $ ctx $ runTypeCheck SkipChecks $ typeCheck x
 
 checkType :: HasType a => TypeEnv -> EffectRow -> a -> Except ()
 checkType env eff x = void $ ctx $ runTypeCheck (CheckWith (env, eff)) $ typeCheck x
-  where ctx = addContext $ "Checking:\n" ++ pprint x
+  where ctx = addContext $ "Checking:\n" ++ pprint x ++ "\nWith env:\n" ++ pprint env
```

Comment:
Hmm I'm a bit scared that printing the `env` will make the error unreadable. WDYT? 😕

Reply:
Sure. I probably just added that for debugging at some point.